include/saber/geometry/point.hpp

```diff
@@ -38,6 +38,10 @@ class Point
     Point& operator*=(const Point& inPoint);
     Point& operator/=(const Point& inPoint);
 
+    // Logical operators
+    bool operator==(const Point& inPoint);
+    bool operator!=(const Point& inPoint);
```

Comment:
You've declared these as "instance methods" rather than "free functions".
This goes against the guidance here:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c86-make--symmetric-with-respect-to-operand-types-and-noexcept

Also be sure to never declare comparison operators, like: `operator==` or `operator!=` as `virtual`:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c87-beware-of--on-base-classes